src/AutoRest.CSharp.V3/ClientModels/ClientBuilder.cs

```diff
@@ -0,0 +1,255 @@
+
+        private static ClientConstant? CreateDefaultValueConstant(Parameter requestParameter) =>
+            requestParameter.ClientDefaultValue != null ?
+                ClientModelBuilderHelpers.ParseClientConstant(requestParameter.ClientDefaultValue, (FrameworkTypeReference)ClientModelBuilderHelpers.CreateType(requestParameter.Schema, requestParameter.IsNullable())) :
+                (ClientConstant?)null;
+
+        private static ClientMethod? BuildMethod(Operation operation)
+        {
+            var httpRequest = operation.Request.Protocol.Http as HttpRequest;
+            var response = operation.Responses.FirstOrDefault();
+            var httpResponse = response?.Protocol.Http as HttpResponse;
+
+            if (httpRequest == null || httpResponse == null)
+            {
+                return null;
+            }
+
+            Dictionary<string, ConstantOrParameter> uriParameters = new Dictionary<string, ConstantOrParameter>();
+            Dictionary<string, PathSegment> pathParameters = new Dictionary<string, PathSegment>();
+            List<QueryParameter> query = new List<QueryParameter>();
+            List<RequestHeader> headers = new List<RequestHeader>();
+
+            List<ServiceClientMethodParameter> methodParameters = new List<ServiceClientMethodParameter>();
+
+            RequestBody? body = null;
+            foreach (Parameter requestParameter in operation.Request.Parameters ?? Array.Empty<Parameter>())
+            {
+                string defaultName = requestParameter.Language.Default.Name;
+                string serializedName = requestParameter.Language.Default.SerializedName ?? defaultName;
+                ConstantOrParameter? constantOrParameter;
+                Schema valueSchema = requestParameter.Schema;
+
+                switch (requestParameter.Schema)
+                {
+                    case ConstantSchema constant:
+                        constantOrParameter = ClientModelBuilderHelpers.ParseClientConstant(constant.Value.Value, ClientModelBuilderHelpers.CreateType(constant.ValueType, constant.Value.Value == null));
+                        valueSchema = constant.ValueType;
+                        break;
+                    case BinarySchema _:
+                        // skip
+                        continue;
+                    //TODO: Workaround for https://github.com/Azure/autorest.csharp/pull/275
+                    case ArraySchema arraySchema when arraySchema.ElementType is ConstantSchema constantInnerType:
+                        constantOrParameter = new ServiceClientMethodParameter(requestParameter.CSharpName(),
+                            new CollectionTypeReference(ClientModelBuilderHelpers.CreateType(constantInnerType.ValueType, false), false),
+                            CreateDefaultValueConstant(requestParameter), false);
+                        break;
+                    //TODO: Workaround for https://github.com/Azure/autorest.csharp/pull/275
```

Comment:
This seems like the wrong link? Or something... I'm not sure how that PR relates to the code that is here now.

Reply:
You've added it :)